ci/test_cpp.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+#!/bin/bash
+
+set -euo pipefail
+
+rapids-logger "Create test conda environment"
+. /opt/conda/etc/profile.d/conda.sh
+
+rapids-dependency-file-generator \
+  --generate conda \
+  --file_key test \
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION}" > env.yaml
+
+rapids-mamba-retry env create --force -f env.yaml -n test
+set +eu
+conda activate test
+set -u
```

Comment:
I'm curious about why these `set` values are used. Can you elaborate?

Reply:
We got unbound errors when running the activate/decativate scripts so I turned off error checking for the conda scripts:

https://github.com/rapidsai/rapids-cmake/actions/runs/3192203924/jobs/5209416886#step:6:392